plugin/core/registry.py

```diff
@@ -156,21 +156,19 @@ def run(self, edit: Any) -> None:
 
 class LspRestartServerCommand(sublime_plugin.TextCommand):
     def is_enabled(self) -> bool:
-        window = self.view.window()
-        lsp_servers = bool()
-        if window:
-            lsp_servers = True if windows.lookup(window).get_sessions() else False
-        return is_supported_view(self.view) and lsp_servers
+        return is_supported_view(self.view) and windows.lookup(self.view.window()).has_active_sessions()
 
     def run(self, edit: Any) -> None:
-        window = self.view.window()
-        if window:
-            self.lsp_server_window = windows.lookup(window)
-            self.lsp_servers = self.lsp_server_window.get_sessions()
-
-            if self.lsp_servers:
-                window.show_quick_panel(self.lsp_servers, self.restart_server)
-
-    def restart_server(self, index) -> None:
-        self.lsp_server_window.restart_session(config_name=self.lsp_servers[index])
+        self.window = self.view.window()
```

Comment:
There is really no need to store anything on the command instance. You can re-retrieve it easily.
And while it's unlikely that the window can change between invocation of the command and `restart_server`, it's still conceptually wrong to store state like that here.